src/matlab2tikz.m

```diff
@@ -2156,7 +2162,10 @@ case size(cData,2) % specific y-data is given
             error('matlab2tikz:UnknownAlphaMapping', ...
                   'Unknown alpha mapping "%s"', alphaMapping);
     end
-    alpha = min(1,max(alpha,0)); % clip at range [0, 1]
+    
+    if isfloat(alpha) %important, alpha data can have integer type which should not be scaled
```

Comment:
Shouldn't the alpha data always be mapped to a float range [0,1] ? With integers, I suppose `0` is invisible and `255` is opaque, but together with L2026 (` hasAlpha = ~all(alphaData(:) == 1);`) this has a different meaning for integers and floats.

I guess you know more about the actual values, so I hope you can enlighten me on what happens in such a case.